src/axom/sidre/examples/spio/IOSCRCkpt.cpp

```diff
@@ -0,0 +1,202 @@
+    // it created the checkpoint in the work loop below.
+    char ckptname[SCR_MAX_FILENAME];
+    SCR_Have_restart(&have_restart, ckptname);
+    if (have_restart) {
+      // Tell SCR we're starting to read the checkpoint.
+      // One must only call SCR_Start_restart if SCR_Have_restart
+      // reports that there is a checkpoint to restart from.
+      // SCR returns the same name in ckptname as SCR_Have_restart.
+      SCR_Start_restart(ckptname);
+
+      // It is common to use the checkpoint name to compute
+      // the path to the checkpoint file.
+      // One should build the path to the file on the parallel
+      // file system.
+      std::string root_file = std::string(ckptname) + "." + file_base + ".root";
+
+      DataStore* ds = new DataStore();
+      SLIC_ASSERT(ds);
+      Group* root = ds->getRoot();
+    
+      IOManager reader(MPI_COMM_WORLD, true);
+      reader.read(root, root_file, false, true);
+    
+      int timestep = root->getView("fields/a/timestep")->getScalar();
+
+      delete ds;
+
+      // Tell SCR whether this process succeeded reading its checkpoint.
+      // Each process should set valid=1 if it read its portion
+      // of the checkpoint successfully and valid=0 otherwise.
+      // SCR executes an allreduce across processes to identify
+      // whether all ranks succeeded.
+      // If any rank failed, SCR considers the checkpoint
+      // to be invalid, and then SCR will load the next most
+      // recent checkpoint if one exists.
+      // The return code will be SCR_SUCCESS if all procs
+      // succeeded.
+      int valid = 1;
+      int rc = SCR_Complete_restart(valid);
+      restarted = (rc == SCR_SUCCESS);
+
+      // If we succeeded, then update initial state
+      if (restarted) {
+        t_start = timestep + 1;
+      }
+    }
+  } while (have_restart && !restarted);
+
+  // Application work loop
+  int t;
+  for (t = t_start; t < t_start + num_ckpts; t++) {
+    /////////////////////////
+    // Do actual work ...
+    /////////////////////////
+
+    // Ask SCR whether it's time to checkpoint is optional.
+    // For checkpoints that are purely defensive, SCR can
+    // help guide the application to the proper checkpoint
+    // frequency for exampled based on checkpoint cost and
+    // expected failure frequency.
+    int need_checkpoint = 0;
+    SCR_Need_checkpoint(&need_checkpoint);
+    if (need_checkpoint) {
+
+      // Tell SCR we're starting a checkpoint.
+      // Each SCR output set should be given a name.
+      // This name should be user-friendly, because
+      // end users may need to type it at a command line.
+      // It should also encode enough information that
+      // one can construct the full path to the file on the
+      // parallel file system.
+      std::string ckptname = "time." + std::to_string(t);
+      SCR_Start_output(ckptname.c_str(), SCR_FLAG_CHECKPOINT);
+
+      DataStore* ds = new DataStore();
```

Comment:
To better match how we'd expect users to use this in practice, the datastore should already exist outside this loop.
It should not be created each cycle.

Reply:
I moved the datastore create/delete outside of the loop